libs/afalg_engine/Makefile

```diff
@@ -0,0 +1,70 @@
+# Copyright (C) 2019 Eneas Ulir de Queiroz <cotequeiroz@gmail.com>
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=afalg_engine
+PKG_VERSION:=1.0.1
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/cotequeiroz/afalg_engine/archive/v$(PKG_VERSION)
+PKG_HASH:=ef3ee1ba3cb4e9145f9a0dea5bc6d3fe6cb7b5b9e68053d474829e84dc1c4988
+
+PKG_MAINTAINER:=Eneas U de Queiroz <cotequeiroz@gmail.com>
+PKG_LICENSE:=Apache-2.0
+PKG_LICENSE_FILES:=LICENSE
+PKG_CONFIG_DEPENDS:= CONFIG_AFALG_ZERO_COPY
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+$(eval $(shell grep "^ENGINES_DIR" "$(TOPDIR)/package/libs/openssl/Makefile"))
+ifeq ($(ENGINES_DIR),)
+  ENGINES_DIR:=engines-1.1
+endif
+
+define Package/libopenssl-afalg_sync
+    SECTION:=libs
+    CATEGORY:=Libraries
+    SUBMENU:=SSL
+    MDEPENDS:=libopenssl
+    TITLE:=AF_ALG engine using sync crypto API
+    URL:=https://github.com/cotequeiroz/afalg_engine
+    DEPENDS:=libopenssl @OPENSSL_ENGINE @!OPENSSL_ENGINE_BUILTIN_AFALG \
```

Comment:
I hate that OPENSSL_ENGINE depend. I hope it works out fine, meaning no recursive dependencies.

Reply:
It does work, don't worry about it.  It wouldn't if I had selected (added a `+` in front of) libopenssl.  You can always see what the dependencies/selections are in `tmp/.config-package.in`:
```
        config PACKAGE_libopenssl-afalg_sync
                tristate "libopenssl-afalg_sync................ AF_ALG engine using sync crypto API"
                default y if DEFAULT_libopenssl-afalg_sync
                default m if ALL
                select PACKAGE_librt if USE_GLIBC
                select PACKAGE_libpthread if USE_GLIBC
                select PACKAGE_libc
                depends on PACKAGE_libopenssl
                select PACKAGE_libopenssl-conf
                select PACKAGE_libssp if GCC_LIBSSP
                depends on OPENSSL_ENGINE
                select PACKAGE_kmod-crypto-user
                depends on !OPENSSL_ENGINE_BUILTIN_AFALG
                depends on PACKAGE_libopenssl
                depends on m || (PACKAGE_libopenssl-afalg != y)
```
If you compare that to the original openssl afalg engine, it's pretty much the same.  The only differences are the guards against installing the both AFALG engines at the same time.   The last line above handles the `CONFLICTS` statement--when the original engine is built separate from libopenssl you can't install both at the same time, or have them both set to `y`; `!OPENSSL_ENGINE_BUILTIN_AFALG` handles the case where the original afalg is built into libopenssl.